okhttp/src/main/java/okhttp3/Authenticator.java

```diff
@@ -71,5 +72,6 @@
    * Returns a request that includes a credential to satisfy an authentication challenge in {@code
    * response}. Returns null if the challenge cannot be satisfied.
    */
-  @Nullable Request authenticate(Route route, Response response) throws IOException;
+  @Nullable Request authenticate(@Nullable Route route,
+                                 @Nonnull Response response) throws IOException;
```

Comment:
Non-null is the default and need not be explicitly annotated

Reply:
Oops, right, thanks. Updated the PR.